src/jit/hwintrinsicxarch.cpp

```diff
@@ -1035,12 +1037,15 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
             retType = getSIMDTypeForSize(retSimdSize);
         }
     }
-    else
+    else if (retType == TYP_STRUCT)
     {
-        assert(retType == TYP_STRUCT);
         baseType = getBaseTypeAndSizeOfSIMDType(sig->retTypeClass, &simdSize);
         retType  = getSIMDTypeForSize(simdSize);
     }
+    else
+    {
+        baseType = getBaseTypeAndSizeOfSIMDType(clsHnd, &simdSize);
```

Comment:
Had to change this to be here as the check right below (`if (!varTypeIsArithmetic(baseType)`) is what handles unsupported `T` and we were returning `nullptr`.

Validated that we now return the integer constant node, that the loop unrolling functionality works (cc. @gfoidl), and the codegen for the known cases is now "efficient".

Reply:
Maybe for 5.0, it would be nice to make the loop unrolling work for "real world" scenarios, such as `for (int i = 0; i < data.Length; i += Vector128<T>.Count)`...

Reply:
Perhaps you could file an issue?

Reply:
Will file one before merging.

Reply:
Looks like this is largely covered by both https://github.com/dotnet/coreclr/issues/11606 and https://github.com/dotnet/coreclr/issues/20486.

I've added comments to both of these instead.